Python/DoublyLinkedList.py

```diff
@@ -0,0 +1,231 @@
+    def removeLast(self):
+        if (self.num == 0):
+            raise Exception("List is empty")
+
+        delNode = self.last.prev
+        out = delNode.item
+        self.last.prev = delNode.prev
+        self.last.prev.next = self.last
+        self.num -= 1
+        return out
+
+    # clears all elements of list
+    def clear(self):
+        self.first = Node(None)
+        self.last = Node(None)
+        self.num = 0
+        self.first.next = self.last
+        self.last.prev = self.first
+
+    # creates and return identical list
+    def clone(self):
+        newLL = DoublyLinkedList()
+        count = 0
+        current = Node(None)
+        current.next = self.first.next
+
+        while (count != self.num):
+            item = current.next.item
+            newLL.addLast(item)
+            current.next = current.next.next
+            count += 1
+
+        return newLL
+
+    # checks if the lsit has the specified element
+    def contains(self, item):
+        if (item == None):
+            raise Exception("Illegal argument")
+
+        current = self.first
+        idx = 0
+
+        while (idx != self.num):
+            current = current.next
+            element = current.item
+            idx += 1
+            if (element == item):
+                return True
+
+        return False
+    
+    # returns first element of the list without removing it
+    def element(self):
+        if (self.num == 0):
+            raise Exception("List is empty")
+        return self.first.next.item
+
+    # returns element at a specified position
+    def get(self, index):
+        idx = 0
+        current = self.first.next
+
+        while (idx != index):
+            current = current.next
+            idx += 1
+
+        item = current.item
+        return item
+
+    # returns first element
+    def getFirst(self):
+        if (self.num == 0):
+            raise Exception("List is empty")
+        return self.first.next.item
+
+    # returns last element
+    def getLast(self):
+         if (self.num == 0):
+             raise Exception("List is empty")
+         return self.last.prev.item
+
+    # returns the index of first occurence of specified element or -1 if no such element
+    def indexOf(self, item):
+        if (self.num == 0):
+            return -1
+        idx = 0
+        current = self.first.next
+
+        while (idx != self.num):
+            currItem = current.item
+            if (currItem == item):
+                return idx
+            current = current.next
+            idx += 1
+
+        return -1
+
+    # iterates the list
+    def iterate(self):
+        if (self.num == 0):
+            raise Exception("List is empty")
+        current = Node(None)
+        current.next = self.first.next
+        while (current.next != self.last):
+            item = current.next.item
+            current.next = current.next.next
+            yield item
+
+# tests
+words = DoublyLinkedList()
```

Comment:
Move all testing code to separate file.
Implement testing using a standard python unit testing framework. 
eg. https://docs.pytest.org/en/stable/